xbmc/addons/kodi-dev-kit/include/kodi/tools/Thread.h

```diff
@@ -0,0 +1,398 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+
+#include "../General.h"
+
+#include <condition_variable>
+#include <future>
+#include <mutex>
+#include <thread>
+
+namespace kodi
+{
+namespace tools
+{
+
+//==============================================================================
+/// @defgroup cpp_kodi_tools_CThread class CThread
+/// @ingroup cpp_kodi_tools
+/// @brief **Helper class to represent threads of execution**\n
+/// An execution thread is a sequence of instructions that can run concurrently
+/// with other such sequences in multithreaded environments while sharing the
+/// same address space.
+///
+/// Is intended to reduce any code work of C++ on addons and to have them faster
+/// to use.
+///
+/// His code uses the support of platform-independent thread system introduced
+/// with C++11.
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/tools/Thread.h>
+/// #include <kodi/AddonBase.h>
+///
+/// class ATTRIBUTE_HIDDEN CTestAddon
+///   : public kodi::addon::CAddonBase,
+///     public kodi::tools::CThread
+/// {
+/// public:
+///   CTestAddon() = default;
+///
+///   ADDON_STATUS Create() override;
+///
+///   void Process() override;
+/// };
+///
+/// ADDON_STATUS CTestAddon::Create()
+/// {
+///   kodi::Log(ADDON_LOG_INFO, "Starting thread");
+///   CreateThread();
+///
+///   Sleep(4000);
+///
+///   kodi::Log(ADDON_LOG_INFO, "Stopping thread");
+///   // This added as example and also becomes stopped by class destructor
+///   StopThread();
+///
+///   return ADDON_STATUS_OK;
+/// }
+///
+/// void CTestAddon::Process()
+/// {
+///   kodi::Log(ADDON_LOG_INFO, "Thread started");
+///
+///   while (!m_threadStop)
+///   {
+///     kodi::Log(ADDON_LOG_INFO, "Hello World");
+///     Sleep(1000);
+///   }
+///
+///   kodi::Log(ADDON_LOG_INFO, "Thread ended");
+/// }
+///
+/// ADDONCREATOR(CTestAddon)
+/// ~~~~~~~~~~~~~
+///
+///@{
+class CThread
```

Comment:
May I suggest to name the class just `Thread`, without the `C`prefix? As far as I know we do not use C prefix for devkit classes.

Reply:
In the main classes I use "C..." only when it came to additional auxiliary data I had left out C prefix (e.g. with PVRChannel or AudioEngineFormat) because there only values are changed but no processes are (so to speak, extended structure).

Reply:
okay. your decision.